src/par_iter/enumerate.rs

```diff
@@ -50,29 +51,50 @@ unsafe impl<M> ExactParallelIterator for Enumerate<M>
 impl<M> IndexedParallelIterator for Enumerate<M>
     where M: IndexedParallelIterator,
 {
-    type Producer = EnumerateProducer<M::Producer>;
+    fn with_producer<WP>(self, wp: WP) -> WP::Output
```

Comment:
Or maybe I am mis-understanding. Anyway, it would be nice if the parameter name was more explicit.